File: solutions/_680_valid_palindrome_ii/solution.py
class Solution:
    def validPalindrome(
        self, s: str, l: int = None, r: int = None, skipped: bool = False
    ) -> bool:
        if skipped is False:
            l = 0
            r = len(s) - 1
        while l < r:
            if s[l] != s[r]:
                if skipped:
                    return False
                return self.validPalindrome(s, l + 1, r, True) or self.validPalindrome(
                    s, l, r - 1, True
                )
            l += 1
            r -= 1
        return True


# https://leetcode.com/problems/valid-palindrome-ii/

File: solutions/_680_valid_palindrome_ii/solution.java
package _680_valid_palindrome_ii;

class Solution {

  public boolean validPalindrome(String s) {
    return validPalindrome(s, 0, s.length() - 1, false);
  }

  public boolean validPalindrome(String s, int l, int r, boolean skipped) {
    while (l < r) {
      if (s.charAt(l) == s.charAt(r)) {
        l += 1;
        r -= 1;
      } else {
        if (skipped) return false;
        return (
          validPalindrome(s, l + 1, r, true) ||
          validPalindrome(s, l, r - 1, true)
        );
      }
    }
    return true;
  }
}
/*
https://leetcode.com/problems/valid-palindrome-ii/
*/
